File: Week2/KimSungHa/B10820.py
# 문제: 10820번
# 설명: 주어진 문자열에 포함된 소문자, 대문자, 숫자, 공백의 갯수를 각각 계산합니다.

while True:
    try:        # 종료 조건을 주지 않았으므로 try except 사용.
        word = input()      # 문자 입력과 체크해야할 갯수 0으로 할당
        lower = 0
        upper = 0
        digit = 0
        vaccancy = 0

        for i in word:          # 문자 전체를 돌며 각각을 찾으면 해당 변수에 +1
            if i.islower():
                lower +=1
            elif i.isupper():
                upper +=1
            elif i.isdigit():
                digit +=1
            elif i == ' ':
                vaccancy +=1

        print(lower, upper, digit, vaccancy)    # f-string없이도 요구 조건에 부합하도록 출력된다.
    except:
        break

File: Week2/gyumin/B10820.java
package gyumin;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class B10820 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        List<String> list = new ArrayList<>();
        String s;
        
        // 개수가 정해지지 않은 n개의 문자열을 입력받으려면 BufferedReader 에 null이 담기지 않을 때까지만 담으면 됨
        // 이클립스 환경 상에서 정상적으로 테스트가 되지 않아 문자열이 비었는지를 따로 확인하고 엔터 입력해서 테스트
        while((s=br.readLine()) != null && !s.isEmpty()) {
        	list.add(s);
        }
        
        for(int i = 0; i < list.size(); i++) {
        	int lower = 0;
        	int upper = 0;
        	int digit = 0;
        	int space = 0;
        	char[] temp = list.get(i).toCharArray();
        	for(int j = 0; j < temp.length; j++) {
        		// 아스키 코드를 이용해서 조건 확인
        		// Character.isLowerCase()
        		if('a' <= temp[j] && temp[j] <= 'z') {
        			lower++;
        		// Character.isUpperCase()
        		} else if('A' <= temp[j] && temp[j] <= 'Z') {
        			upper++;
        		// Character.isDigit
        		} else if('0' <= temp[j] && temp[j] <= '9') {
        			digit++;
        		// 공백 확인
        		} else if(temp[j] == ' ') {
        			space++;
        		}
        	}
        	
        	System.out.printf("%d %d %d %d\n", lower, upper, digit, space);
        }
        
        br.close();
    }
}
